Source/objects.cpp

```diff
@@ -2073,24 +2030,46 @@ void OperateL5LDoor(Object &door, bool sendflag)
 	}
 }
 
-void OperateL1Door(const Player &player, Object &door)
+void OperateDoor(const Player &player, Object &door)
 {
 	int dpx = abs(door.position.x - player.position.tile.x);
 	int dpy = abs(door.position.y - player.position.tile.y);
-	if (dpx == 1 && dpy <= 1 && door._otype == OBJ_L1LDOOR)
-		OperateL1LDoor(door, true);
-	if (dpx <= 1 && dpy == 1 && door._otype == OBJ_L1RDOOR)
-		OperateL1RDoor(door, true);
-}
-
-void OperateL5Door(const Player &player, Object &door)
-{
-	int dpx = abs(door.position.x - player.position.tile.x);
-	int dpy = abs(door.position.y - player.position.tile.y);
-	if (dpx == 1 && dpy <= 1 && door._otype == OBJ_L5LDOOR)
-		OperateL5LDoor(door, true);
-	if (dpx <= 1 && dpy == 1 && door._otype == OBJ_L5RDOOR)
-		OperateL5RDoor(door, true);
+	if (dpx == 1 && dpy <= 1) {
+		switch (door._otype) {
+		case OBJ_L1LDOOR:
+			OperateL1LDoor(door, true);
+			break;
+		case OBJ_L2LDOOR:
+			OperateL2LDoor(door, true);
+			break;
+		case OBJ_L3RDOOR:
+			OperateL3RDoor(door, true);
+			break;
+		case OBJ_L5LDOOR:
+			OperateL5LDoor(door, true);
+			break;
+		default:
+			break;
+		}
+	}
+	if (dpx <= 1 && dpy == 1 && door._otype == OBJ_L1RDOOR) {

```

Comment:
```suggestion
	if (dpx <= 1 && dpy == 1) {
```

Reply:
Must have pressed undo one to many times after trying to push teleflag in there :)